nemo/core/optim/mcore_optim.py

```diff
@@ -55,8 +55,13 @@ def state_dict(self):
     def load_state_dict(self, state_dict):
         self.mcore_optimizer.load_state_dict(state_dict)
 
-    def sharded_state_dict(self, model_sharded_state_dict, is_loading: bool = False, **kwargs):
-        return self.mcore_optimizer.sharded_state_dict(model_sharded_state_dict, is_loading, **kwargs)
+    def sharded_state_dict(self, model_sharded_state_dict, optimizer_state_dict=None, is_loading=False, **kwargs):
+        sharding_type = (
+            'fully_sharded_bucket_space' if kwargs.get('dist_ckpt_parallel_save', False) else 'dp_zero_gather_scatter'
```

Comment:
what's the link between `kwargs['dist_ckpt_parallel_save']` and the yaml config?

Reply:
did you tests running both sharding types?

Reply:
Updated this part, PTAL at the latest part. Thanks